cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -409,6 +413,79 @@ func (s *Supervisor) startOpAMP() error {
 	return nil
 }
 
+func (s *Supervisor) stopOpAMP() error {
+	s.logger.Debug("Stopping OpAMP client...")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	err := s.opampClient.Stop(ctx)
+	// TODO(srikanthccv): remove context.DeadlineExceeded after https://github.com/open-telemetry/opamp-go/pull/213
+	if err != nil && !errors.Is(err, context.DeadlineExceeded) {
```

Comment:
This must be removed after the graceful shutdown gets merged. Wanted to get the reviews so skipping this temporarily.